src/com/palmergames/bukkit/towny/command/TownCommand.java

```diff
@@ -709,10 +735,23 @@ public void townToggle(Player player, String[] split) throws TownyException {
 
 			} else if (split[0].equalsIgnoreCase("pvp")) {
 				// Make sure we are allowed to set these permissions.
-				toggleTest(player, town, StringMgmt.join(split, " "));
-				town.setPVP(!town.isPVP());
-				TownyMessaging.sendTownMessage(town, String.format(TownySettings.getLangString("msg_changed_pvp"), town.getName(), town.isPVP() ? "Enabled" : "Disabled"));
-
+				boolean outsiderintown = false;

```

Comment:
Needs an 
if(TownySettings.configsettinghere) {

}
so that the new code only runs when the new config option is true. (It should be made false by default to keep normal pvp toggling in play)

The comments of the new config option should state that using this feature may cause a bit of lag.

Reply:
Maybe GTOWN_SETTINGS_OUTSIDERS_PREVENT_PVP_TOGGLE